pyproject.toml

```diff
@@ -31,7 +31,7 @@ classifiers = [
 dynamic=["version"]
 [project.optional-dependencies]
 dev = [
-    "ruff == 0.2.1",
+    "ruff",
```

Comment:
I chose to unpin this here, maybe I shouldn't have.

My thinking was:
- The commit hook has its own checkout of ruff, so the developer's version **could** diverge, though they may see different errors if they manually run ruff.
- But I liked that oddity more than I liked developers having to manage different packages wanting different ruff versions.
- Thoughts?

Reply:
Hmm, though then what version should the CI install...? Should that be hardcoded in the CI to match the hook? Or should we hardcode the version here in the pyproject.toml... hmmm.... Maybe I should leave this alone. But I don't love it.

Reply:
I started doing explicit pins on these in response to a SQLFluff linting error that caused malformed code in some cases. I think the astral folks are less likely to make the same mistake, but i sort of like 'a human has decided to increment the version on the autoformatting tools'. could be talked out of this.